### PR TITLE
DOC: special: complete documentation of `eval_*` functions

### PR DESCRIPTION
**Pull request by person142**
Adds parameters, returns, and see also sections; makes clear what
happens when the degree is not integral.

### CODE REVIEW

File: scipy/special/add_newdocs.py [hunk 1]
```diff
@@ -1392,111 +1392,568 @@ def add_newdoc(place, name, doc):
...
     eval_jacobi(n, alpha, beta, x, out=None)
 
     Evaluate Jacobi polynomial at a point.
+
+    The Jacobi polynomials can be defined via the Gauss hypergeometric
+    function :math:`{}_2F_1` as
+
+    .. math::
+
+        P_n^{(\alpha, \beta)}(x) = \frac{(\alpha + 1)_n}{\Gamma(n + 1)}
```
Comment (ev-br) [inline scipy/special/add_newdocs.py]: Trivial nit: maybe also define `alpha_n`

File: scipy/special/add_newdocs.py [hunk 2]
```diff
@@ -1392,111 +1392,568 @@ def add_newdoc(place, name, doc):
...
+    U : ndarray
+        Values of the Chebyshev polynomial
+
+    See Also
+    --------
+    u_roots : roots and quadrature weights of Chebyshev polynomials of
+              the second kind
+    chebyu : Chebyshev polynomial object
+    eval_chebyt : evaluate Chebyshev polynomials of the first kind
+    hyp2f1 : Gauss hypergeometric function
```
Comment (ev-br) [inline scipy/special/add_newdocs.py]: Also link to numpy polynomials, here and elsewhere where applicable?

File: scipy/special/add_newdocs.py [hunk 3]
```diff
@@ -1392,111 +1392,568 @@ def add_newdoc(place, name, doc):
...
 
-    Evaluate normalized Hermite polynomial at a point.
+    Evaluate probabilist's (normalized) Hermite polynomial at a
+    point.
+
+    Defined by
+
+    .. math::
+
+        He_n(x) = (-1)^ne^{x^2/2}\frac{d^n}{dx^n}e^{-x^2/2};
```
Comment (ev-br) [inline scipy/special/add_newdocs.py]: Optional, trivial: maybe throw in some spaces for those who read the text docstrings? Also to counter the argument that "TeX is not readable"

**Review by ev-br — COMMENTED** [review]
A big +1 from me. I left a couple of trivial comments, all of which are optional.

**Comment by person142** [comment]
Made the fixes and added a couple more see also links.

**Comment by ev-br** [comment]
It might be useful to keep doing separate `GEN:` commits to separate autogeneration of code (`$ python generate_ufuncs.py` etc). That way it's a little more readable. But that's a minor detail.

You mentioned somewhere that you have a series of PRs in mind.  So I'll merge this now to keep the ball rolling. Thanks Josh!

**Pull request merged**